CMakeLists.txt

```diff
@@ -174,14 +175,44 @@ if(BUILD_TESTING)
     --output-junit testresults.xml
     --output-on-failure
   )
-endif(BUILD_TESTING)
+endif(BUILD_TESTING AND NOT IS_SUBDIRECTORY)
 
 include(FindPkgConfig)
 
 # Libaries in deps but built separately
-find_package(QCBOR REQUIRED)
+add_subdirectory(deps/QCBOR)
```

Comment:
I don't want to inherit external project build rules in this project. If it is helpful to have in-tree builds of these deps it should be done by its own option. And if QCBOR targets have bad config that needs to be handled with an upstream issue on QCBOR.